internal/pkg/build/sources/conveyorPacker_zypper.go

```diff
@@ -74,6 +79,124 @@ func (cp *ZypperConveyorPacker) Get(b *types.Bundle) (err error) {
+	updateurl, updateurlOk := cp.b.Recipe.Header["updateurl"]
+	sleproduct, sleproductOk := cp.b.Recipe.Header["product"]
+	sleuser, sleuserOk := cp.b.Recipe.Header["user"]
+	sleregcode, sleregcodeOk := cp.b.Recipe.Header["regcode"]
+	slepgp, slepgpOk := cp.b.Recipe.Header["productpgp"]
+	sleurl, sleurlOk := cp.b.Recipe.Header["registerurl"]
+	slemodules, slemodulesOk := cp.b.Recipe.Header["modules"]
+	cnt := -1
+	if tmp, ok := cp.b.Recipe.Header["otherurl0"]; ok {
+		otherurl[0] = tmp
+		cnt = 1
+	} else {
+		if tmp, ok := cp.b.Recipe.Header["otherurl1"]; ok {
+			otherurl[0] = tmp
+			cnt = 2
+		}
+	}
+	for i := 1; cnt > 0 && i < 20; i++ {
+		numS := strconv.Itoa(cnt)
+		if tmp, ok := cp.b.Recipe.Header["otherurl"+numS]; ok {
+			otherurl[i] = tmp
+			cnt++
+		} else {
+			cnt = -1
+		}
+	}
+	regex := regexp.MustCompile(`(?i)%{OSVERSION}`)
+
+	if sleproductOk || sleuserOk || sleregcodeOk {
+		if !sleproductOk || !sleuserOk || !sleregcodeOk {
+			return fmt.Errorf("for installation of SLE 'Product', 'User' and 'Regcode' need to be set")
+		}
+		if !osversionOk {
+			return fmt.Errorf("invalid zypper header, OSVersion always required for SLE")
+		}
+		if !slepgpOk && !mirrorurlOk {
+			return fmt.Errorf("no 'SLEpgp' and no 'InstallURL' defined in bootstrap definition")
+		}
+		suseconnectPath, err = exec.LookPath("SUSEConnect")
+		if err != nil {
+			return fmt.Errorf("SUSEConnect is not in PATH: %v", err)
+		}
+
+		array := strings.SplitN(osversion, ".", -1)
+		osmajor := array[0]
+		iosmajor, err = strconv.Atoi(osmajor)
+		if err != nil {
+			return fmt.Errorf("OSVersion has wrong format %v", err)
+		}
+		osminor := ""
+		if len(array) > 1 {
+			osminor = "." + array[1]
+		}
+		if iosmajor > 12 && !mirrorurlOk {
+			return fmt.Errorf("for SLE version > 12 'MirrorURL' must be defined and point to the installer")
+		}
+		osservicepack := ""
+		tmp, err := strconv.Atoi(array[1])
+		if err != nil {
+			return fmt.Errorf("cannot convert minor version string to integer: %v", err)
+		}
+		if len(array) > 1 && tmp > 0 {
+			osservicepack = array[1]
+		}
+		if mirrorurlOk {
+			mirrorurl = regex.ReplaceAllString(mirrorurl, osmajor+osservicepack)
+		}
+		sleproduct = regex.ReplaceAllString(sleproduct, osmajor+osservicepack)
+		array = strings.SplitN(sleproduct, "/", -1)
+		machine, _ := machine()
+		if len(array) == 3 {
+			machine = array[2]
+		}
+		suseconnectProduct = sleproduct
+		suseconnectModver = osmajor + osminor + "/" + machine
+		switch len(array) {
+		case 1:
+		case 2:
+			suseconnectProduct += "/" + machine
+		case 3:
+			suseconnectProduct += "/" + osversion + "/" + machine
+		default:
+			return fmt.Errorf("malformed Product setting")
+		}
+		if slepgpOk {
+			tmpfile, err := ioutil.TempFile("/tmp", "singularity-pgp")
+			if err != nil {
+				return fmt.Errorf("cannot create pgp-file: %v", err)
+			}
+			pgpfile = tmpfile.Name()
+
+			if _, err = tmpfile.WriteString(slepgp); err != nil {
```

Comment:
I ran into this error when trying to import the gpg key from the example:
```
ian@linux-of0y:~/singularity> sudo singularity build test.sif examples/sle/Singularity 
[sudo] password for root: 
Build target already exists. Do you want to overwrite? [N/y] y
INFO:    Starting build...
error: /tmp/singularity-pgp233218452: key 1 not an armored public key.
FATAL:   While performing build: conveyor failed to get: while importing pgp keys: exit status 1
```

Looks like this was caused by a missing newline at the end og `slepgp`. 

There appears to be two ways to fix this:
1. always append a newline here
2. add `\n` to the last line of the key in the definition and change the parser to always replace any trailing `\n` characters with a newline character

Reply:
Indeed, I had not tested this part any more after I had done changes. As soon as a MirrorImage is provided (which is the case for SLE-15 upwards), this key is no longer required. While retesting this I came across some other issues as well. All have been addressed in my latest push.